src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symbol.java

```diff
@@ -2276,6 +2277,21 @@ public Type getReceiverType() {
             return asType().getReceiverType();
         }
 
+        public Type implicitReceiverType() {
+            ClassSymbol enclosingClass = enclClass();
+            if (enclosingClass == null) {
+                return null;
+            }
+            Type enclosingType = enclClass().type;
+            if (isConstructor()) {
+                return enclosingType.getEnclosingType();
+            }
+            if (!isStatic()) {
```

Comment:
Just curious, what happens in such a scenario:
```java
public class Outer {
  public static void work() {
    class Local {
      Local() {} // <- what's the implicitReceiverType here?
    }
  }
}
```
FYI javac currently is not marking such local classes in static methods as `static` in its `InnerClasses` attribute.

Reply:
Good question! I debugged that example, and `getEnclosingType()` returns `NoType`, which I think is what we want.

This is the logic that initializes the enclosing type and handles local classes:

https://github.com/openjdk/jdk/blob/c788160f8acea7b58b54ad857b601bb7ffb53f8e/src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Enter.java#L508-L519